interfaces/builtin/content_test.go

```diff
@@ -225,13 +225,24 @@ apps:
 
 func (s *ContentSuite) TestResolveSpecialVariable(c *C) {
 	info := snaptest.MockInfo(c, "{name: name, version: 0}", &snap.SideInfo{Revision: snap.R(42)})
-	c.Check(builtin.ResolveSpecialVariable("foo", info), Equals, filepath.Join(dirs.CoreSnapMountDir, "name/42/foo"))
 	c.Check(builtin.ResolveSpecialVariable("$SNAP/foo", info), Equals, filepath.Join(dirs.CoreSnapMountDir, "name/42/foo"))
 	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA/foo", info), Equals, "/var/snap/name/42/foo")
 	c.Check(builtin.ResolveSpecialVariable("$SNAP_COMMON/foo", info), Equals, "/var/snap/name/common/foo")
 	c.Check(builtin.ResolveSpecialVariable("$SNAP", info), Equals, filepath.Join(dirs.CoreSnapMountDir, "name/42"))
 	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA", info), Equals, "/var/snap/name/42")
 	c.Check(builtin.ResolveSpecialVariable("$SNAP_COMMON", info), Equals, "/var/snap/name/common")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP//", info), Equals, filepath.Join(dirs.CoreSnapMountDir, "name/42")+"//")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA/", info), Equals, "/var/snap/name/42/")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP_DATA//", info), Equals, "/var/snap/name/42//")
+	// automatically prefixed with $SNAP
+	c.Check(builtin.ResolveSpecialVariable("foo", info), Equals, filepath.Join(dirs.CoreSnapMountDir, "name/42/foo"))
+	// contain valid $SNAP_* variables, but starting with one of them, end
+	// up automatically prefixed with $SNAP
+	c.Check(builtin.ResolveSpecialVariable("foo/snap/$SNAP/bar", info), Equals, "/snap/name/42/foo/snap//snap/name/42/bar")
+	c.Check(builtin.ResolveSpecialVariable("$SNAP/foo/snap/$SNAP/bar", info), Equals, "/snap/name/42/foo/snap//snap/name/42/bar")
```

Comment:
This looks like we are testing to make sure something that went wrong is predictably weird. Why are you testing for `/snap/name/42/foo/snap//snap/name/42/bar` when it, AFAIK, should never happen in practice? I'm asking here and above because I feel I am missing a larger context.

Reply:
Dropped it, the test case was unnecessarily confusing.